pandas/core/indexes/timedeltas.py

```diff
@@ -154,32 +153,73 @@ def __new__(cls, data=None, unit=None, freq=None, start=None, end=None,
             result.name = name
             return result
 
-        if unit is not None:
-            data = to_timedelta(data, unit=unit, box=False)
-
         if is_scalar(data):
-            raise ValueError('TimedeltaIndex() must be called with a '
+            raise ValueError('{cls}() must be called with a '
                              'collection of some kind, {data} was passed'
-                             .format(data=repr(data)))
+                             .format(cls=cls.__name__, data=repr(data)))
 
-        # convert if not already
-        if getattr(data, 'dtype', None) != _TD_DTYPE:
+        if isinstance(data, TimedeltaIndex) and freq is None and name is None:
+            if copy:
+                return data.copy()
+            else:
+                return data._shallow_copy()
+
+        # - Cases checked above all return/raise before reaching here - #
+
+        if unit is not None:
             data = to_timedelta(data, unit=unit, box=False)
-        elif copy:
-            data = np.array(data, copy=True)
-
-        data = np.array(data, copy=False)
-        if data.dtype == np.object_:
-            data = array_to_timedelta64(data)
-        if data.dtype != _TD_DTYPE:
-            if is_timedelta64_dtype(data):
+
+        # Unwrap whatever we have into a np.ndarray
+        if not hasattr(data, 'dtype'):
+            # e.g. list, tuple
+            if np.ndim(data) == 0:
+                # i.e.g generator
+                data = list(data)
+            data = np.array(data, copy=False)
+        elif isinstance(data, ABCSeries):
+            data = data._values
+        elif isinstance(data, (cls, TimedeltaArrayMixin)):
+            data = data._data
+
+        # Convert whatever we have into timedelta64[ns] dtype
+        if is_object_dtype(data) or is_string_dtype(data):
+            # no need to make a copy, need to convert if string-dtyped
```

Comment:
because we have several tests that specifically pass e.g. `np.array(['1 days', '4 days'], dtype='<S6')`.  (BTW we don't have similar tests for DatetimeIndex, which I intend to fix in the DTI analogue of this PR)